ocaml/xenops/xenops.ml

```diff
@@ -228,7 +228,7 @@ let stat ~xs d =
 		| x -> string_of_kind x in
 	let frontend_type = match d.frontend.kind with
 		| Vbd _ | Tap ->
-			let be = frontend_path_of_device ~xs d in
+			let be = frontend_ro_path_of_device ~xs d in
 			(try if xs.Xs.read (sprintf "%s/device-type" be) = "cdrom" then "cdrom" else "disk" with _ -> "??")
```

Comment:
As for the place where we read from the `rw` path, yes, this is to read the frontend state (just a number) from the path that the guest can write, so that the guest can tell us the state.